src/pybind/mgr/dashboard/module.py

```diff
@@ -221,6 +222,67 @@ def await_configuration(self):
                 self.log.info("Configured CherryPy, starting engine...")  # type: ignore
                 return uri
 
+    def configure_cors(self):
+        """
+        Allow CORS requests if the cross_origin_url option is set.
+        """
+        cross_origin_url = mgr.get_localized_module_option('cross_origin_url', '')
+        if cross_origin_url:
+            cherrypy.tools.CORS = cherrypy.Tool('before_handler', self.cors_tool)
+            config = {
+                'tools.CORS.on': True,
+            }
+            self.update_cherrypy_config(config)
+
+    def cors_tool(self):
+        '''
+        Handle both simple and complex CORS requests
+
+        Add CORS headers to each response. If the request is a CORS preflight
+        request swap out the default handler with a simple, single-purpose handler
+        that verifies the request and provides a valid CORS response.
+        '''
+        req_head = cherrypy.request.headers
+        resp_head = cherrypy.response.headers
+
+        # Always set response headers necessary for 'simple' CORS.
+        req_header_origin_url = req_head.get('Access-Control-Allow-Origin')
+        cross_origin_urls = mgr.get_localized_module_option('cross_origin_url', '')
+        cross_origin_url_list = [url.strip() for url in cross_origin_urls.split(',')]
```

Comment:
Every request handler will be executed once per request (and requests are one of the scaling factors of webservers). Therefore it's essential to keep them simple and remove any processing that it's strictly not mandatory. The fact that every HTTP request requires to interact with the ceph-mgr API might result in a DoS vulnerability (unauthenticated users might send thousands of dummy HTTP requests and cause a performance degradation in the ceph-mgr). I remember that `get_module_option` calls are cached and don't end up hitting the `ceph-mon`, but even though it's healthy not to risk that.

My recommendation here would be to remove the `get_localized_module_option()` (BTW, `get_module_option()` would be enough, since there's no reason to configure different CORS setting per active manager) and pass that fixed param to the `cors_tool(self, cross_origin_list)` and then configured as `'tools.cors_tool.cross_origin_list': '...'`.